module/enrichers.mjs

```diff
@@ -823,6 +903,36 @@ async function rollAction(event) {
         case "tool":
           options.ability = ability;
           return await actor.rollToolCheck(tool, options);
+        case "item":
+          if ( target.dataset.rollItemUuid ) {
+            const gameActor = await fromUuid(target.dataset.rollItemActor);
+            if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+              return (await fromUuid(target.dataset.rollItemUuid)).use();
+            } else {
+              return ui.notifications.warn(`You do not have ownership of ${gameActor?.name}, and cannot use this item.`);
+            }
+          } else if ( target.dataset.rollItemName ) {
+            if ( target.dataset.rollItemActor ) {
+              const gameActor = await fromUuid(target.dataset.rollItemActor);
+              if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+                if ( canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName) ) {
+                  return canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName).use();
+                } else if ( gameActor.items.getName(target.dataset.rollItemName) ) {
+                  await gameActor.items.getName(target.dataset.rollItemName).use();
+                  if (canvas.tokens.controlled.length > 0 ) {
+                    ui.notifications.warn(`Your controlled token ${canvas.tokens.controlled[0].name} does not have an Item with name ${target.dataset.rollItemName} using Item Enricher owner`);
```

Comment:
This warning seems misplaced?

Reply:
We already checked if the user has a token selected, but without the Named item, so this is to give feedback that we arent rolling the item from that selected token but from the enricher owner, and its only needed if they actually have a token selected, as this is also the method that would be used if they dont have anything selected. Where else should it go?

Reply:
I would expect a warning that my selected token's actor simply doesn't have the item, and then nothing more happens.

Reply:
This is what we discussed up here https://github.com/foundryvtt/dnd5e/pull/2785#issuecomment-1965119618